index.js

```diff
@@ -0,0 +1,91 @@
+const express = require('express');
+const knex = require('knex')
+const session = require('express-session');
+const KnexSessionStore = require('connect-session-knex')(session)
+const knexConfig = require('./knexfile.js')
+const db = knex(knexConfig.development)
+const bcrypt = require('bcryptjs')
+const server = express();
+
+const sessionConfig = {
+  name: 'monkey',
+  secret: 'hello world',
+  cookie: {
+    maxAge: 1000 * 60 * 20 * 2,
+    secure: false,
+  },
+  httpOnly: true,
+  resave: false,
+
+  saveUninitialized: false,
+  //this stores the cookie data on the database. So you can restart the server
+  store: new KnexSessionStore({
+    tablename: 'sessions',
+    sidfieldname: 'sid',
+    knex: db,
+    createtable: true,
+    clearInterval: 1000 * 60 * 60,
+  }),
+}
+
+server.use(session(sessionConfig))
+server.use(express.json())
+
+server.get('/', (req, res) => {
+  res.send(`Hello users number: ${req.session.user}`)
+})
+
+protected = (req, res, next) => {
+  if(req.session && req.session.user) {
+    next()
+  } else {
+    res.status(401).json({message: 'You shall not pass!'})
+  }
+}
+
+server.get('/api/users', protected, (req, res) => {
+      db('users')
+        .then(users => res.status(200).json(users))
+        .catch(err => res.status(500).json(err))
+})
+
+server.post('/api/users', (req, res) => {
```

Comment:
That route is a little misleading. Better aproach would be to call it `register`.